build.gradle

```diff
@@ -1,145 +1,171 @@
-licenseFile = rootProject.file('LICENSE.txt')
-// copyright notices
-noticeFile = rootProject.file('NOTICE.txt')
+    skipIntegTests = System.getProperty("skip.integtests", "false")
+  }
 
-// disable uploadArchives task for now, no upload happening currently
-//GSI TODO: disabled for now
-//uploadArchives.enabled = false
+  repositories {
+    mavenLocal()
+    maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
+    mavenCentral()
+    maven { url "https://plugins.gradle.org/m2/" }
+  }
 
-opensearchplugin {
-  name 'ltr'
-  description 'Learning to Rank Query w/ RankLib Models'
-  classname 'com.o19s.es.ltr.LtrQueryParserPlugin'
-  // license of the plugin, may be different than the above license
-  licenseFile = rootProject.file('LICENSE.txt')
-  // copyright notices, may be different than the above notice
-  noticeFile = rootProject.file('NOTICE.txt')
+  dependencies {
+    classpath "org.opensearch.gradle:build-tools:${opensearch_version}"
+  }
 }
 
-// In this section you declare the dependencies for your production and test code
-// OpenSearch dependency is included due to the build-tools, test-framework as well
 repositories {
-  mavenCentral()
   mavenLocal()
+  maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
+  mavenCentral()
+  maven { url "https://plugins.gradle.org/m2/" }
 }
 
 dependencies {
+  implementation "com.github.spullara.mustache.java:compiler:0.9.3"
+  implementation "org.opensearch:opensearch:${opensearch_version}"
+  implementation "org.opensearch:opensearch-x-content:${opensearch_version}"
+  implementation "org.opensearch:opensearch-common:${opensearch_version}"
+  implementation "org.opensearch:opensearch-core:${opensearch_version}"
+  implementation "org.opensearch:common-utils:${plugin_version}"
   implementation "org.apache.lucene:lucene-expressions:${luceneVersion}"
+  implementation 'com.o19s:RankyMcRankFace:0.1.1'
   implementation "org.antlr:antlr4-runtime:${antlrVersion}"
   implementation "org.ow2.asm:asm:${ow2Version}"
   implementation "org.ow2.asm:asm-commons:${ow2Version}"
   implementation "org.ow2.asm:asm-tree:${ow2Version}"
-  implementation 'com.o19s:RankyMcRankFace:0.1.1'
-  implementation "com.github.spullara.mustache.java:compiler:0.9.3"
-
-  //testImplementation 'org.apache.lucene:lucene-test-framework:${luceneVersion}'
-  runtimeOnly 'org.locationtech.spatial4j:spatial4j:0.7'
-  runtimeOnly 'org.locationtech.jts:jts-core:1.15.0'
-  runtimeOnly 'org.apache.logging.log4j:log4j-core:2.17.1'
 }
 
 
-
-
-// see https://github.com/opensearch-project/OpenSearch/blob/0ba0e7cc26060f964fcbf6ee45bae53b3a9941d0/buildSrc/src/main/java/org/opensearch/gradle/precommit/DependencyLicensesTask.java
-dependencyLicenses {
-  mapping from: /lucene-.*/, to: 'lucene'
-  mapping from: /asm-.*/, to: 'asm'
-  mapping from: /RankyMcRankFace.*/, to: 'lucene'
-  //mapping from: /Ranky.*/, to: 'lucene'
-  mapping from: /compiler.*/, to: 'lucene'
-}
-
-sourceSets {
```

Comment:
Hi @macohen 

I am trying to understand why have we replaced the sourceSets with [this] (https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/14/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R151)

<details>
<summary>The change</summary>
```
sourceSets {
  main {
    resources {
      srcDirs = ["config"]
      includes = ["**/*.yml"]
    }
  }
}
```
</details>


I am unable to run the plugin locally (both debug and normal mode) due to this change
It is unable to access the mapping file `"fstore-index-mapping.json"`

<details>
<summary>Exact eror</summary>
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_state_exception",
                "reason": "failed to create ltr feature store index with resource [fstore-index-mapping.json]"
            }
        ],
        "type": "illegal_state_exception",
        "reason": "failed to create ltr feature store index with resource [fstore-index-mapping.json]",
        "caused_by": {
            "type": "null_pointer_exception",
            "reason": null
        }
    },
    "status": 500
}
```
</details>

Without this change I am able to create `.ltrstore`

I might have missed details, I am starting to get my hands dirty

Reply:
The PR was merged